lua/lspconfig/util.lua

```diff
@@ -21,6 +21,21 @@ M.default_config = {
 -- global on_setup hook
 M.on_setup = nil
 
+---@param filetype string the filetype to check (can be a compound, dot-separated filetype; see |'filetype'|)
```

Comment:
this new function has no docstring explaining its purpose. and this PR has a wall of text discussing it.

how in the world is anyone supposed know when to use this

Reply:
apologies, I didn't realize that it was unclear. I've added documentation. let me know if it's unclear

Reply:
doesn't look like there is a summary docstring. also a couple input => output examples would go a long way.

```
filetype "foo.bar" matches expected "..."
```